### PR TITLE
Enabled ESRPCodeSign for docfx v2 - dev

### PR DESCRIPTION
**Pull request by sharanya-rao**
[AB#201338](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/201338)

### CODE REVIEW

File: pack.ps1 [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+        Exit 1
+    }
+}
+
+function ValidateCommand {
+    param($command)
+    return (Get-Command $command -ErrorAction SilentlyContinue) -ne $null
+}
+
+# Check if dotnet cli exists globally
```
Comment (superyyrrzz) [inline pack.ps1:57]: Can we extract `prepare.ps1` to avoid duplicate code with `build.ps`?

File: tools/Deployment/gulpfile.js [hunk 2]
```diff
@@ -245,7 +250,8 @@ gulp.task("syncBranchCore", () => {
 });
 gulp.task("test", gulp.series("clean", "build", "e2eTest", "publish:myget-test"));
 gulp.task("dev", gulp.series("clean", "build", "e2eTest"));
-gulp.task("dev:release", gulp.series("clean", "build", "e2eTest", "publish:myget-dev", "publish:azdevops-perf-login", "publish:azdevops-perf", "publish:azdevops-internal-login", "publish:azdevops-internal", "publish:azdevops-ppe-login", "publish:azdevops-ppe"));
+gulp.task("dev:release", gulp.series("clean", "build", "e2eTest"));
```
Comment (superyyrrzz) [inline tools/Deployment/gulpfile.js:253]: We can remove this task as it is same as the above one. Rename to `dev:build`.

Comment (sharanya-rao) [reply tools/Deployment/gulpfile.js:253]: They will not be same, I am updating.

Comment (superyyrrzz) [reply tools/Deployment/gulpfile.js:253]: Then, update below one to `dev:release`. It includes pack and publish.

In the next PR, we can use `master:build`, `master:pack`, `master:release` (only for publish).

File: pack.ps1 [hunk 3]
```diff
@@ -0,0 +1,165 @@
+param([string] $configuration = "Release")
+
+$ErrorActionPreference = 'Stop'
+$framework = "net472"
+$packageVersionFilePath = ".\package_version_temp.txt" # build.ps1 saves the package version to this temp file
+$packageVersion = Get-Content -Path $packageVersionFilePath
```
Comment (superyyrrzz) [inline pack.ps1:6]: Check existence before `Get-Content`. Hint user to run `build.ps` if it does not exists.

File: pack.ps1 [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    };
+    "MarkdownMigrateTool"       = @{
+        "proj"    = $null;
+        "nuspecs" = @("src/nuspec/MarkdownMigrateTool/MarkdownMigrateTool.nuspec");
+    };
+    "YamlSplitter"              = @{
+        "proj"    = $null;
+        "nuspecs" = @("src/nuspec/YamlSplitter/YamlSplitter.nuspec");
+    };
+	"SandcastleRefMapper"       = @{
```
Comment (superyyrrzz) [inline pack.ps1:102]: nit: indent with spaces

File: tools/Deployment/gulpfile.js [hunk 5]
```diff
@@ -243,9 +248,10 @@ gulp.task("syncBranchCore", () => {
     let docfxHome = path.resolve(config.docfx.home);
     return SyncBranch.runAsync(repoUrl, docfxHome, config.sync.fromBranch, config.sync.targetBranch);
 });
-gulp.task("test", gulp.series("clean", "build", "e2eTest", "publish:myget-test"));
-gulp.task("dev", gulp.series("clean", "build", "e2eTest"));
-gulp.task("dev:release", gulp.series("clean", "build", "e2eTest", "publish:myget-dev", "publish:azdevops-perf-login", "publish:azdevops-perf", "publish:azdevops-internal-login", "publish:azdevops-internal", "publish:azdevops-ppe-login", "publish:azdevops-ppe"));
+gulp.task("test", gulp.series("clean", "build", "pack", "e2eTest", "publish:myget-test"));
```
Comment (superyyrrzz) [inline tools/Deployment/gulpfile.js:251]: I am not sure whether we still use `test` task. If used, we can put `pack` after `e2eTest`. We only need to pack them when e2eTest runs successfully. Same for `dev:release`.

File: tools/Deployment/gulpfile.js [hunk 6]
```diff
@@ -243,9 +248,10 @@ gulp.task("syncBranchCore", () => {
...
+gulp.task("dev", gulp.series("clean", "build", "pack", "e2eTest"));
```
Comment (superyyrrzz) [inline tools/Deployment/gulpfile.js:252]: This is used for PR build for validation. No need to `pack` here.

File: codesign_policy.json [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+              {
+                "parameterName": "OpusName",
+                "parameterValue": "Microsoft"
+              },
+              {
+                "parameterName": "OpusInfo",
+                "parameterValue": "http://www.microsoft.com"
+              },
+              {
+                  "parameterName": "Append",
```
Comment (superyyrrzz) [inline codesign_policy.json:90]: nit: indent - 2

File: codesign_policy.json [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+            ],
+            "ToolName": "sign",
+            "ToolVersion": "1.0"
+          }
+        ]
+      }
+    },
+    {
+      "MatchedPath": [
+          "DotLiquid.resources.dll",
```
Comment (superyyrrzz) [inline codesign_policy.json:57]: I think you used `!Microsoft*.dll` in this PR earlier. Can it work?

Comment (sharanya-rao) [reply codesign_policy.json:57]: Doesn't seem to work, therefore explicitly listed them.

**Pull request merged**